### PR TITLE
[mips/uk-ua] Add ukrainian translation for MIPS Assemly

### PR DESCRIPTION
**Pull request by AstiaSun**
- [x] I solemnly swear that this is all original content of which I am the original author
- [x] Pull request title is prepended with `[language/lang-code]`
- [x] Pull request touches only one file (or a set of logically related files with similar changes made)
- [x] Content changes are aimed at *intermediate to experienced programmers* (this is a poor format for explaining fundamental programming concepts)
- [x] If you've changed any part of the YAML Frontmatter, make sure it is formatted according to [CONTRIBUTING.md](https://github.com/adambard/learnxinyminutes-docs/blob/master/CONTRIBUTING.markdown)
  - [x] Yes, I have double-checked quotes and field names!

### CODE REVIEW

File: uk-ua/mips-ua.html.markdown [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+    lb $t2, label                           # Скопіювати буквений символ в частину
+                                            # нижчого порядку регістра $t2
+    lb $t2, 0($s0)                          # Скопіювати буквений символ з адреси
+                                            # в $s0 із зсувом 0
+    # Подібне використання і 'lh' для halfwords
+
+    sw $t0, label                           # Зберегти чисельне значення в адресу в 
+                                            # пам'яті, що відповідає змінній label
+    sw $t0, 8($s0)                          # Зберегти чисельне значення в адресу, 
+                                            # зазначеній у $s0, та зі зсувом у 8 байтів
```
Comment (Menelion) [inline uk-ua/mips-ua.html.markdown]: адресу, зазначену?

File: uk-ua/mips-ua.html.markdown [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+    li $t1, 6
+    add $t2, $t0, $t1                       # $t2 = $t0 + $t1
+    sub $t2, $t0, $t1                       # $t2 = $t0 - $t1
+    mul $t2, $t0, $t1                       # $t2 = $t0 * $t1
+    div $t2, $t0, $t1                       # $t2 = $t0 / $t1 (Може не підтримуватись
+                                            # деякими версіями MARS)
+    div $t0, $t1                            # Виконує $t0 / $t1. Отримати частку можна
+                                            # за допомогою команди 'mflo', остаток - 'mfhi'
+
+    # Bitwise Shifting
```
Comment (Menelion) [inline uk-ua/mips-ua.html.markdown]: Бітовий зсув?

File: uk-ua/mips-ua.html.markdown [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+    srl $t0, $t0, 5                         # Побітовий здвиг вправо (не зберігає 
+                                            # знак, знак розширюється 0)
+    srlv $t0, $t1, $t2                      # Здвиг вправо зі змінною кількістю у
+                                            # регістрі
+    sra $t0, $t0, 7                         # Побітовий арифметичний здвиг вправо 
+                                            # (зберігає знак)
+    srav $t0, $t1, $t2                      # Здвиг вправо зі змінною кількістю у 
+                                            # регістрі
+
+    # Bitwise operators
```
Comment (Menelion) [inline uk-ua/mips-ua.html.markdown]: Бітові оператори

File: uk-ua/mips-ua.html.markdown [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+    and $t0, $t1, $t2                       # Побітове І (AND)
+    andi $t0, $t1, 0xFFF                    # Побітове І з безпосереднім значенням
+    or $t0, $t1, $t2                        # Побітове АЛЕ (OR)
```
Comment (Menelion) [inline uk-ua/mips-ua.html.markdown]: Або? :)

File: uk-ua/mips-ua.html.markdown [hunk 5]
```diff
@@ -0,0 +1,366 @@
...
+    ori $t0, $t1, 0xFFF                     # Побітове АЛЕ з безпосереднім значенням
```
Comment (Menelion) [inline uk-ua/mips-ua.html.markdown]: Або

File: uk-ua/mips-ua.html.markdown [hunk 6]
```diff
@@ -0,0 +1,366 @@
...
+## Розгалуження ##
+  _branching:
+    # В основному інструкції розгалуження мають наступну форму:
+    # <instr> <reg1> <reg2> <label>
+    # де label - це назва змінної, в яку ми хочемо перейти, якщо зазначене твердження
+    # правдиве
+
+    beq $t0, $t1, reg_eq                    # Перейдемо у розгалуження reg_eq
+                                            # якщо $t0 == $t1, інакше -
+                                            # виконати наступну строку
```
Comment (Menelion) [inline uk-ua/mips-ua.html.markdown]: наступний рядок

File: uk-ua/mips-ua.html.markdown [hunk 7]
```diff
@@ -0,0 +1,366 @@
...
+    #   if (a > c)
+    #     max = a;
+    #   else
+    #     max = c;
+    # else
+    #     max = b;
+    #   else
+    #     max = c;
+
+    # Нехай $s0 = a, $s1 = b, $s2 = c, $v0 = повернути регіст
```
Comment (Menelion) [inline uk-ua/mips-ua.html.markdown]: регістр

File: uk-ua/mips-ua.html.markdown [hunk 8]
```diff
@@ -0,0 +1,366 @@
...
+      move $v0, $s2                         # max = c
+
+    done:                                   # Кінець програми
+
+## Цикли ##
+  _loops:
+    # Цикл складається з умови виходу та з інструкції переходу після його завершення
+    li $t0, 0
+    while:
+      bgt $t0, 10, end_while                # Коли $t0 менше 10, продовжувати інтеграції
```
Comment (Menelion) [inline uk-ua/mips-ua.html.markdown]: ітерації

File: uk-ua/mips-ua.html.markdown [hunk 9]
```diff
@@ -0,0 +1,366 @@
...
+      li $v0, 10
+      syscall
+    
+    # Ця функція повертає 1
+    return_1:
+      li $v0, 1                           # Завантажити val в регіст $v0
+      jr $ra                              # Повернутись до попереднього ПЦ і продовжити виконання 
+
+
+    # Function with 2 args
```
Comment (Menelion) [inline uk-ua/mips-ua.html.markdown]: Функція з двома аргументами

File: uk-ua/mips-ua.html.markdown [hunk 10]
```diff
@@ -0,0 +1,366 @@
...
+        jr $ra
+
+## Макроси ##
+  _macros:
+    # Макроси надзвичайно корисні для заміни блоків коду, що повторюються, за допомогою
+    # однієї змінної, для покращення читабельності
+    # Це не заміна функцій.
+    # Вони мають бути оголошені перед використанням
+
+    # Макрос для виведення нових строк (оскільки операція досить часто виконується)
```
Comment (Menelion) [inline uk-ua/mips-ua.html.markdown]: рядків?

File: uk-ua/mips-ua.html.markdown [hunk 11]
```diff
@@ -0,0 +1,366 @@
...
+    .macro println()
+      la $a0, newline                     # Значення нової строки зберігатиметься тут
```
Comment (Menelion) [inline uk-ua/mips-ua.html.markdown]: нового рядка

File: uk-ua/mips-ua.html.markdown [hunk 12]
```diff
@@ -0,0 +1,366 @@
...
+    .macro print_int(%num)
+      li $v0, 1
+      lw $a0, %num
+      syscall
+    .end_macro
+    
+    li $t0, 1
+    print_int($t0)
+    
+    # Ми таком можемо передавати безпосередньо значення в макроси
```
Comment (Menelion) [inline uk-ua/mips-ua.html.markdown]: також

File: uk-ua/mips-ua.html.markdown [hunk 13]
```diff
@@ -0,0 +1,366 @@
...
+    print(hello_world)
+
+## Масиви ##
+.data
+  list: .word 3, 0, 1, 2, 6                 # Це масив чисел
+  char_arr: .asciiz "hello"                 # Це текстовий масив
+  buffer: .space 128                        # Виділяє блок пам'яті, що
+                                            # автоматично не очищується
+                                            # Ці блоки пам'яті вирівнюють
+                                            # один одного
```
Comment (Menelion) [inline uk-ua/mips-ua.html.markdown]: Гм... These blocks of memory are aligned next to each other — Ці блоки пам’яти вирівнені поруч один з одним?.. Якось так)

File: uk-ua/mips-ua.html.markdown [hunk 14]
```diff
@@ -0,0 +1,366 @@
...
+
+.text
+  la $s0, list                              # Завантажити адрегу у список
```
Comment (Menelion) [inline uk-ua/mips-ua.html.markdown]: адресу списку

**Comment by AstiaSun** [comment]
@Menelion Thanks so so much for the review! I have updated the translation.

**Comment by Menelion** [comment]
@AstiaSun Sorry, but some conversations are still shown as unresolved...

**Comment by AstiaSun** [comment]
@Menelion Hmm, I see all comments as outdated. I have resolved all outdated conversations.

**Comment by Menelion** [comment]
@AstiaSun Thanks, it's OK now! :)

**Pull request merged**